src/app/donation-start/donation-start-form/donation-start-form.component.ts

```diff
@@ -1740,17 +1816,22 @@ export class DonationStartFormComponent implements AfterContentChecked, AfterCon
     }
   }
 
-  private clearDonation(donation: Donation, clearAllRecord: boolean) {
+  /**
+   * Also resets captcha & relevant donation persistence state mgmt,
+   * and returns to step 1 so required input can be collected again.
+   * We DON'T reset `this.donor`, so there should be no need for a new captcha code.
+   *
+   * @param jumpToStart If the caller is already setting up a known-value donation alongside the clear,
+   *                        this should be false. In other cases we need to know the new amount so it
+   *                        should usually be true, if the page is not being unloaded.
+   */
+  private clearDonation(donation: Donation, {clearAllRecord, jumpToStart}: {clearAllRecord: boolean; jumpToStart: boolean}) {
     if (clearAllRecord) { // i.e. don't keep donation around for /thanks/... or reuse.
```

Comment:
```suggestion
   * @param clearAllRecord Don't keep donation around for /thanks/... or reuse.
   * @param jumpToStart If the caller is already setting up a known-value donation alongside the clear,
   *                        this should be false. In other cases we need to know the new amount so it
   *                        should usually be true, if the page is not being unloaded.
   */
  private clearDonation(donation: Donation, clearAllRecord: boolean, jumpToStart: boolean) {
    if (clearAllRecord) {
```

Reply:
I think this suggestion is reversing the destructure bit. Don't think I can partially apply it so will try to mimic locally

Reply:
ah yeah sorry that was a mistake, I just meant to move the comment about the meaning of `clearAllRecord` into the docblock